pennylane_qiskit/noise_models.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module contains functions for converting Qiskit NoiseModel objects
+into PennyLane NoiseModels.
+"""
+from collections import defaultdict
+from typing import Tuple
+from warnings import warn
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import AnyWires
+from qiskit.quantum_info.operators.channel import Kraus
+
+# pylint:disable = protected-access
+qiskit_op_map = {
+    "x": "X",
+    "y": "Y",
+    "z": "Z",
+    "h": "Hadamard",
+    "cx": "CNOT",
+    "cz": "CZ",
+    "swap": "SWAP",
+    "iswap": "ISWAP",
+    "rx": "RX",
+    "ry": "RY",
+    "rz": "RZ",
+    "id": "Identity",
+    "cswap": "CSWAP",
+    "crx": "CRX",
+    "cry": "CRY",
+    "crz": "CRZ",
+    "p": "PhaseShift",
+    "ccx": "Toffoli",
+    "qubitunitary": "QubitUnitary",
+    "u1": "U1",
+    "u2": "U2",
+    "u3": "U3",
+    "rzz": "IsingZZ",
+    "ryy": "IsingYY",
+    "rxx": "IsingXX",
+    "s": "S",
+    "t": "T",
+    "sx": "SX",
+    "cy": "CY",
+    "ch": "CH",
+    "cp": "CPhase",
+    "ccz": "CCZ",
+    "ecr": "ECR",
+    "sdg": qml.adjoint(qml.S),
+    "tdg": qml.adjoint(qml.T),
+    "sxdg": qml.adjoint(qml.SX),
+    "reset": qml.measure(AnyWires, reset=True),  # TODO: Improve reset support
+}
+
+
+def _build_qerror_op(error, **kwargs) -> qml.QubitChannel:
+    """Builds a PennyLane error channel from a Qiksit's QuantumError object.
+
+    Args:
+        error (QuantumError): Quantum error object
+        kwargs: Optional keyword arguments used for conversion
+
+    Returns:
+        qml.QubitChannel: an equivalent PennyLane channel operation
+    """
+    try:
+        kraus_matrices = np.round(Kraus(error).data, decimals=kwargs.get("decimals", 10))
+    except Exception as exc:  # pragma: no cover
+        raise ValueError(f"Error {error} could not be converted.") from exc
+
+    return qml.QubitChannel(K_list=kraus_matrices, wires=AnyWires)
+
+
+def _build_noise_model_map(noise_model, **kwargs) -> Tuple[dict, dict]:
+    """Builds a noise model map from which a PennyLane noise model can be constructed efficiently.
```

Comment:
```suggestion
    """Builds a noise model map from a Qiskit noise model. This noise model map can be used to efficiently construct a PennyLane noise model.
```

Maybe this is better? Just want to clarify that the function takes in a qiskit noise model.